src/code/z_map_disp.c

```diff
@@ -1,109 +1,1877 @@
+            texLRY += 4;
+            texULY -= 4;
+            gSPTextureRectangle(POLY_OPA_DISP++, texULX << 2, texULY << 2, texLRX << 2, (texLRY) << 2, G_TX_RENDERTILE,
+                                0, 0, 0x036E, 0x02AA);
+        } else {
+            gSPTextureRectangle(POLY_OPA_DISP++, (texULX + 1) << 2, texULY << 2, (texLRX - 1) << 2, texLRY << 2,
+                                G_TX_RENDERTILE, 0, 0, 1 << 10, 1 << 10);
+        }
+        Gfx_SetupDL42_Opa(play->state.gfxCtx);
+
+        CLOSE_DISPS(play->state.gfxCtx);
+    }
+}
+
+s32 MapDisp_IsValidStorey(s32 storey) {
+    if ((sMapDisp.mapDataScene == NULL) || (sSceneNumRooms == 0)) {
+        return false;
+    }
+    if ((storey < 0) || (storey > 5)) {
+        return false;
+    }
+    if (sMapDisp.storeyYList[storey] != FLOOR_MIN_Y) {
+        return true;
+    }
+    return false;
+}
+
+s32 MapDisp_GetPlayerStorey(s16 checkY) {
+    s32 i;
+
+    if ((sMapDisp.mapDataScene == NULL) || (sSceneNumRooms == 0)) {
+        return -1;
+    }
+    if (sMapDisp.numStoreys <= 1) {
+        return 0;
+    }
+    if ((sMapDisp.storeyYList[1] - 5) >= checkY) {
+        return 0;
+    }
+    for (i = 1; i < sMapDisp.numStoreys; i++) {
+        if (((sMapDisp.storeyYList[i] - 5) < checkY) && ((sMapDisp.storeyYList[i + 1] - 5) >= checkY)) {
+            return i;
+        }
+    }
+    return sMapDisp.numStoreys - 1;
+}
+
+typedef struct {
+    /* 0x0 */ s16 sceneId;
+    /* 0x4 */ s32 offsetX;
+    /* 0x8 */ s32 offsetY;
+} MapCustomPosOffset; // size = 0xC
+
+/**
+ * Draws the dungeon map within the pause menu's Map screen.
+ *
+ * @param play
+ */
+void MapDisp_DrawDungeonMap(PlayState* play) {
+    static MapCustomPosOffset sCustomMapOffset[] = {
+        { SCENE_MITURIN, 0, -10 },
+        { SCENE_MITURIN_BS, 0, -10 },
+    };
+    MapDataRoom* mapDataRoom;
+    f32 scaleFrac;
+    s32 scale;
+    s32 var_v0;
+    s32 dungeonIndex = 0;
+    s32 offsetX = 0;
+    s32 offsetY = 0;
+
+    if (MapDisp_SkipDrawDungeonMap(play)) {
+        return;
+    }
+
+    if (Map_IsInBossScene(play)) {
+        switch (play->sceneId) {
+            case SCENE_MITURIN_BS:
+                dungeonIndex = DUNGEON_INDEX_WOODFALL_TEMPLE;
+                break;
+
+            case SCENE_HAKUGIN_BS:
+                dungeonIndex = DUNGEON_INDEX_SNOWHEAD_TEMPLE;
+                break;
+
+            case SCENE_SEA_BS:
+                dungeonIndex = DUNGEON_INDEX_GREAT_BAY_TEMPLE;
+                break;
+
+            case SCENE_INISIE_BS:
+                dungeonIndex = DUNGEON_INDEX_STONE_TOWER_TEMPLE;
+                break;
+
+            default:
+                break;
+        }
+    } else {
+        dungeonIndex = gSaveContext.mapIndex;
+    }
+
+    mapDataRoom = sMapDisp.mapDataScene->rooms;
+    if ((mapDataRoom->mapId == MAP_DATA_NO_MAP) || (mapDataRoom->mapId >= MAPDATA_MAP_GRAND_MAX)) {
+        return;
+    }
+
+    var_v0 = MapData_GetMapCompactId(mapDataRoom->mapId);
```

Comment:
```suggestion
    compactId= MapData_GetMapCompactId(mapDataRoom->mapId);
```

Reply:
v0 is reused later in the function as an unrelated iterator, suggesting that there is another temp, but I couldn't figure out how to add another variable to the stack.

Reply:
Ah I missed that. Yeah keep it unnamed for now I guess.